pyro/poutine/indep_messenger.py

```diff
@@ -53,7 +71,46 @@ def next_context(self):
         """
         self.counter += 1
 
+    def __enter__(self):
+        if self._vectorized is not False:
+            self._vectorized = True
+
+        if self._vectorized is True:
```

Comment:
I'd rather leave it explicit like this, since we're using three values (`None`, `False`, `True`) in the `_vectorized` state machine rather than just two.